### PR TITLE
tests: several improvements to the nested suite

### PR DESCRIPTION
**Pull request by fgimenez**
The improvements include:
  * logic simplified
  * outer and inner spread have differentiated output

Also, the tests executed in the nested suite is restricted due to CI time limitations.

### CODE REVIEW

File: tests/nested/image-build/task.yaml [hunk 1]
```diff
@@ -2,58 +2,73 @@ summary: create ubuntu-core image and execute the suite in a nested qemu instanc
...
-    #teardown_bridge $TAP_DEVICE
+    rm -rf /tmp/work-dir
 
 execute: |
-    . $TESTSLIB/nested.sh
+    execute_remote(){
+        sshpass -p ubuntu ssh -p 8022 -q -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no user1@localhost "$*"
+    }
+
+    prepare_ssh(){
```
Comment (niemeyer) [inline tests/nested/image-build/task.yaml]: These two functions would be better off as comments on top of their logic instead. Using them only once and even defining them in the opposite order than they are called below seems to obfuscate what is going on unnecessarily.

File: tests/nested/image-build/task.yaml [hunk 2]
```diff
@@ -2,58 +2,73 @@ summary: create ubuntu-core image and execute the suite in a nested qemu instanc
...
-                        external:ubuntu-core-16-64:tests/main/install-store \
-                        external:ubuntu-core-16-64:tests/main/interfaces-firewall-control \
-                        external:ubuntu-core-16-64:tests/main/interfaces-system-observe \
-                        external:ubuntu-core-16-64:tests/main/ubuntu-core-classic
+    set +x
+    export SPREAD_EXTERNAL_ADDRESS=localhost:8022
+    /tmp/work-dir/spread -v external:ubuntu-core-16-64:tests/main/ubuntu-core-reboot \
+                            external:ubuntu-core-16-64:tests/main/install-store \
+                            external:ubuntu-core-16-64:tests/main/interfaces-system-observe \
+                            external:ubuntu-core-16-64:tests/main/op-remove-retry | while read line; do echo "> $line"; done
```
Comment (niemeyer) [inline tests/nested/image-build/task.yaml]: This might be `| sed 's/^/> /'`

File: tests/nested/image-build/task.yaml [hunk 3]
```diff
@@ -2,58 +2,73 @@ summary: create ubuntu-core image and execute the suite in a nested qemu instanc
...
     # determine arch related vars
-    set_vars
+    case "$NESTED_ARCH" in
+    amd64)
+        QEMU="$(which qemu-system-x86_64)"
+        ;;
+    i386)
+        QEMU="$(which qemu-system-i386)"
+        ;;
+    *)
```
Comment (zyga) [inline tests/nested/image-build/task.yaml]: I wonder if our armhf bits could be tested this way?

File: tests/nested/image-build/task.yaml [hunk 4]
```diff
@@ -2,58 +2,73 @@ summary: create ubuntu-core image and execute the suite in a nested qemu instanc
...
-    rm -rf $WORK_DIR
-
-    #. $TESTSLIB/nested.sh
-    #teardown_bridge $TAP_DEVICE
+    rm -rf /tmp/work-dir
 
 execute: |
-    . $TESTSLIB/nested.sh
+    execute_remote(){
+        sshpass -p ubuntu ssh -p 8022 -q -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no user1@localhost "$*"
```
Comment (zyga) [inline tests/nested/image-build/task.yaml]: Where is the `user1@` coming from? is that in the assertion?

File: tests/nested/image-build/task.yaml [hunk 5]
```diff
@@ -2,58 +2,73 @@ summary: create ubuntu-core image and execute the suite in a nested qemu instanc
...
 
     prepare_ssh
 
     cd $SPREAD_PATH
 
-    curl -s -O https://niemeyer.s3.amazonaws.com/spread-amd64.tar.gz && tar xzvf spread-amd64.tar.gz && rm -f spread-amd64.tar.gz && mv spread $WORK_DIR
+    curl -s -O https://niemeyer.s3.amazonaws.com/spread-amd64.tar.gz && tar xzvf spread-amd64.tar.gz && rm -f spread-amd64.tar.gz && mv spread /tmp/work-dir
 
-    $WORK_DIR/spread -v external:ubuntu-core-16-64:tests/main/ubuntu-core-reboot \
-                        external:ubuntu-core-16-64:tests/main/install-sideload \
```
Comment (zyga) [inline tests/nested/image-build/task.yaml]: Not sure what the context is, why is `install-sideload` and `ubuntu-core-classic` removed and `op-remove-retry` added?

Comment (niemeyer) [inline tests/nested/image-build/task.yaml]: These are arbitrarily chosen tests. We need to reduce the overall run time of the test so it fits in the 50 minutes Travis gives us.

File: tests/nested/image-build/task.yaml (hunk 1 above)
Comment (fgimenez) [inline tests/nested/image-build/task.yaml]: Thanks, removed

File: tests/nested/image-build/task.yaml (hunk 2 above)
Comment (fgimenez) [inline tests/nested/image-build/task.yaml]: I tried sed but didn't work, it seemed to buffer all the lines of the output and do the change when there were no more of them, that would kill us on travis.

File: tests/nested/image-build/task.yaml (hunk 3 above)
Comment (fgimenez) [inline tests/nested/image-build/task.yaml]: Indeed, that was the original idea and what would give us more value from all this approach, I've spent some time with the pi2 emulator with no luck so far, will keep trying anyway :)

File: tests/nested/image-build/task.yaml (hunk 4 above)
Comment (fgimenez) [inline tests/nested/image-build/task.yaml]: Exactly https://github.com/snapcore/snapd/blob/master/tests/lib/assertions/auto-import.assert#L16

**Pull request merged**